### PR TITLE
feat(23750): consume custom amount

### PR DESCRIPTION
**Pull request by nnorbert**
Get custom amount from Forms submission data and send it to Payments.

### CODE REVIEW

File: apps/journey-api/src/libraries/integratorEngine/index.ts [hunk 1]
```diff
@@ -42,8 +42,8 @@ class IntegratorEngine {
     this.logtoUserId = props.logtoUserId;
   }
 
-  public executeStep(data: StepDataDO): Promise<{ url: string }> {
-    return this.plugin.execute(data, {
+  public executeStep(data: StepDataDO, submissionData: SubmissionData[]): Promise<{ url: string }> {
```
Comment (alfonsograziano) [inline apps/journey-api/src/libraries/integratorEngine/index.ts:45]: shall we rename `submissionData` in `runData` for coherence? And maybe, `runData` might be an object with the entire context of the current run (as we might need run information, with a field for the `runSteps`. Might be an idea for you?

File: apps/journey-api/src/libraries/integratorEngine/plugins/paymentPlugin.ts [hunk 2]
```diff
@@ -8,8 +8,21 @@ import { PaymentRunStepData } from "../../../routes/schemas";
 
 export class PaymentPlugin implements IIntegratorPlugin {
   private readonly paymentLink = "/paymentRequest/pay";
+
+  private getCustomAmount(submissionData: SubmissionData[]) {
+    const formSubmission = submissionData.find((submission) =>
```
[the remark this answers is not in the document]
Comment (nnorbert) [reply apps/journey-api/src/libraries/integratorEngine/plugins/paymentPlugin.ts:13]: Definitely, if you have to get a specific step-related value, you must know the reference to it. We can have a separate investigation, as you mentioned, sure.

File: apps/journey-api/src/libraries/integratorEngine/index.ts (hunk 1 above)
Comment (nnorbert) [reply apps/journey-api/src/libraries/integratorEngine/index.ts]: Done

**Review by alfonsograziano — APPROVED** [review]
LGTM

**Pull request merged**